R/as-data-frame.R

```diff
@@ -73,3 +73,49 @@ as.data.frame.CrunchDataFrame <- function (x, row.names = NULL, optional = FALSE
     names(out) <- names(x)
     return(structure(out, class="data.frame", row.names=c(NA, -nrow(x))))
 }
+
+
+#' as.data.frame method for VariableCatalog
+#'
+#' This method allows you to import the VariableCatalog into your R sessions in 
+#' order to facilliate interactive use. Modifying the dataframe produced by this
+#' method will not update the Crunch web app, for information about modifying 
+#' variables in the app see: \code{vingette("variables",
+#' package = "crunch")}  
+#' 
+#' @param x
+#' A VariableCatalog produced by \code{variables(ds)}.
+#' @param ... Additional arguments passed to \code{data.frame}
+#' @param row.names part of as.data.frame signature. Ignored.
+#' @param optional part of as.data.frame signature. Ignored.
+#' @param fields 
+#' A character vector of the fields from the variable catalog which you would like 
+#' included in the datafame. To include all fields or see which fields are available
+#' set fields to "all". 
+#' @return A data frame
+#' @examples 
+#' \dontrun{
+#' ds <- loadDataset("iris")
+#' vars <- variables(ds)
+#' var_df <- as.data.frame(vars, fields = "all")
+#' }
+#' 
+#' @rdname VariableCatalog-to-Data-Frame
+#' @export
+as.data.frame.VariableCatalog <- function(x, 
+    row.names = NULL, 
+    optional = FALSE, 
+    ...,
+    fields = c("alias", "name", "type")) {
+    out <- catalogToDataFrame(x, rownames = NA)
```

Comment:
I want a reasonable behavior that works for all Catalogs so that we don't have to reimplement the method for each Catalog unless we want to specialize it. 

See https://github.com/Crunch-io/rcrunch/blob/master/R/categories.R#L197-L201 for a model of how this kind of validation is handled elsewhere (grep for `serialPaste` if you want to see lots more). You can put something like this in `catalogToDataFrame`, have the error message include the `class()` of the object passed in, and if you want to list all of the names of the valid fields, feel free. So the error message I am picturing would look like "'banana' is not a valid field for class VariableCatalog.", and, should you choose, followed by "Valid fields are [list of fields]", though I don't require the list of fields.